src/Makefile.inc

```diff
@@ -0,0 +1,340 @@
+############################################
+
+############################################
+#this section includes build targets       #
+############################################
+all : $(ALL)
+
+####################################
+# recompile if the Makefile changes#
+####################################
+#$(OBJS) $(PROTO_OBJS) ($PROTO_EXTRA_OBJS): Makefile target/tx/$(FAMILY)/$(TARGET)/Makefile.inc $(EXTRA_MAKEFILES)
+
+##################################################################################
+# The following enables quiet output unless you use VERBOSE=1                    #
+# Note that this must be after the 1st rule so that it doesn't execute by default#
+##################################################################################
+$(VERBOSE).SILENT:
+
+.PHONY: clean language lint
+
+clean:
+	rm -f $(TARGET).$(EXEEXT) $(TARGET).exe $(TARGET).bin $(TARGET).dfu $(TARGET).list \
+		$(TARGET).map $(ODIR)/*.o $(ODIR)/*.o_ $(ODIR)/*.P  $(ODIR)/*.bin \
+		filesystem/$(FILESYSYTEM) 2> /dev/null || true
+
+language:
+	../utils/extract_strings.pl -po > $(SDIR)/fs/language/deviation.po
+
+
+zip: $(TARGET).zip
+
+%.zip: $(ALL) $(TARGET).dfu $(PROTO_MODULES)
+	#This is not an emulator build (emulator is hanled in target/common/emu/Makefile.inc)
+	cp -f $(TARGET).dfu deviation-$(HGVERSION).dfu
+	rm -f deviation-$(HGVERSION).zip
+	rm -f debug-$(HGVERSION).zip
+	zip debug-$(HGVERSION).zip $(TARGET).elf
+	# $(MAKE) zipping more compatible with other distributions
+	export p=`pwd`; cd $(ODIR) && test -e devo.fs && zip -u $$p/deviation-$(HGVERSION).zip devo.fs && rm devo.fs || true
+	zip deviation-$(HGVERSION).zip deviation-$(HGVERSION).dfu UPDATING.md
+ifdef MODULAR
+	export p=`pwd`; cd $(ODIR) && zip -ur $$p/deviation-$(HGVERSION).zip protocol/*.mod
+	export p=`pwd`; cd $(ODIR) && zip -u $$p/debug-$(HGVERSION).zip *.elf
+	rm -rf $(ODIR)/protocol
+endif
+ifeq "$(INCLUDE_FS)" "1"
+	export p=`pwd`; cd filesystem/$(FILESYSTEM) && zip -ur $$p/deviation-$(HGVERSION).zip  *
+endif
+	zip -u deviation-$(HGVERSION).zip debug-$(HGVERSION).zip
+	rm -f debug-$(HGVERSION).zip
+	rm -f deviation-$(HGVERSION).dfu
+
+fs: $(TARGET).fs
+
+%.fs:
+	rm -f deviation-fs-$(HGVERSION).zip; zip -r deviation-fs-$(HGVERSION).zip filesystem/$(FILESYSTEM)
+
+release:  $(TXS:%=zip_%) $(EMUS:%=zip_win_%) $(TXS:%=fs_%)
+
+coverity:
+	rm -rf cov-int 2>/dev/null; $(MAKE) distclean
+	cov-build --dir cov-int make TARGET=devo8
+	cov-build --dir cov-int make TARGET=devo10
+	tar -cvzf coverity-$(HGVERSION).tgz cov-int
+
+lint:
+	$(SDIR)/../utils/run_linter.py --diff --debug
+
+##########################################
+#Ensure necessary directories are created#
+##########################################
+$(OBJS): | $(ODIR)
+
+$(ODIR):
+	@mkdir -p $@
+
+$(MODELDIR):
+	@mkdir -p $@
+
+######################
+#The main executable #
+######################
+$(TARGET).$(EXEEXT): $(LINKFILE) $(OBJS) $(LIBOPENCM3)
+	@echo " + Building '$@'"
```

Comment:
libopencm3 stuff is target related.  it really doesn't belong in the main makefile.  except that it currently lives in the src/ directory.  maybe it should go into a libs/ directory or under target somewhere...I dunno

Reply:
sure. we can. we need first define the MACROs and document somewhere. then the code can be much clean.